metrics/repl.lag/lag.go

```diff
@@ -115,70 +130,122 @@ func (c *Lag) Help() blip.CollectorHelp {
 	}
 }
 
+// Prepare prepares lag collectors for all levels in the plan that contain the "repl.lag" domain
 func (c *Lag) Prepare(ctx context.Context, plan blip.Plan) (func(), error) {
 LEVEL:
-	for _, level := range plan.Levels {
+	for levelName, level := range plan.Levels {
 		dom, ok := level.Collect[DOMAIN]
 		if !ok {
 			continue LEVEL // not collected in this level
 		}
+		cleanup, err := c.prepareLevel(levelName, plan.MonitorId, plan.Name, dom.Options)
+		if err != nil {
+			return cleanup, err
+		}
+	}
+	return nil, nil
+}
 
-		table := dom.Options[OPT_TABLE]
-		if table == "" {
-			table = blip.DEFAULT_HEARTBEAT_TABLE
+func (c *Lag) Collect(ctx context.Context, levelName string) ([]blip.MetricValue, error) {
+	switch c.lagSourceIn[levelName] {
+	case LAG_SOURCE_BLIP:
+		return c.collectBlipHeartBeatLag(ctx, levelName)
+	case LAG_SOURCE_PFS:
+		return c.collectLagFromPFS(ctx, levelName)
+	}
+
+	panic(fmt.Sprintf("invalid lag source in Collect %s", c.lagSourceIn[levelName]))
+}
+
+// //////////////////////////////////////////////////////////////////////////
+// Internal methods
+// //////////////////////////////////////////////////////////////////////////
+
+func (c *Lag) prepareLevel(levelName string, monitorID string, monitorName string, options map[string]string) (func(), error) {
```

Comment:
Hmm, I know Dan usually recommend against using named returns, so I have been avoiding it. But I can see how this can help with readability. But to be consistent with other parts of code base I would like to hear @daniel-nichter 's thoughts before going ahead with this change.